src/MoBi.Presentation/Presenter/BaseDiagram/MoBiBaseDiagramPresenter.cs

```diff
@@ -102,13 +98,16 @@ public PointF CurrentInsertLocation
       /// </summary>
       public void Handle(EntitySelectedEvent eventToHandle)
       {
-         if (DiagramManager == null) return;
-         if (!DiagramManager.MustHandleExisting(eventToHandle.ObjectBase.Id)) return;
+         if (DiagramManager == null)
+            return;
 
-         IBaseNode baseNode = DiagramModel.GetNode(eventToHandle.ObjectBase.Id);
+         if (!DiagramManager.MustHandleExisting(eventToHandle.ObjectBase.Id))
```

Comment:
We could handle this event without ObjectBase here, but I prevent the event from being raised in the first place.